dnf/cli/commands/updateinfo.py

```diff
@@ -322,96 +322,56 @@ def type2label(typ, sev):
-            # sort filenames
-            if advlist[9]:
-                advlist[9] = sorted(advlist[9])
-            # installed value
-            if not mixed:
-                advlist[10] = None
-            else:
-                advlist[10] = [_('true') if advlist[10] else _('false')]
-            return advlist
-
+        arches = self.base.sack.list_arches()
+        verbose = self.base.conf.verbose
         labels = (_('Update ID'), _('Type'), _('Updated'), _('Bugs'),
                   _('CVEs'), _('Description'), _('Severity'), _('Rights'),
                   _('Files'), _('Installed'))
-        width = _maxlen(labels)
-        for advlist in self._info(apkg_adv_insts):
-            print('=' * 79)
-            print('  ' + advlist[0])
-            print('=' * 79)
-            for label, lines in zip(labels, advlist2lines(advlist)[1:]):
-                if lines is None or lines == [None]:
+
+        def advisory2info(advisory, installed):
+            attributes = [
+                [advisory.id],
+                [self.TYPE2LABEL.get(advisory.type, _('unknown'))],
+                [unicode(advisory.updated)],
+                [],
+                [],
+                (advisory.description or '').splitlines(),
+                [advisory.severity],
+                (advisory.rights or '').splitlines(),
+                sorted(set(pkg.filename for pkg in advisory.packages
+                           if pkg.arch in arches)),
+                None]
+            for ref in advisory.references:
+                if ref.type == hawkey.REFERENCE_BUGZILLA:
+                    attributes[3].append('{} - {}'.format(ref.id, ref.title or ''))
+                elif ref.type == hawkey.REFERENCE_CVE:
+                    attributes[4].append(ref.id)
+            attributes[3].sort()
+            attributes[4].sort()
+            if not verbose:
+                attributes[7] = None
+                attributes[8] = None
+            if mixed:
+                attributes[9] = [_('true') if installed else _('false')]
+
+            width = _maxlen(labels)
+            lines = []
+            lines.append('=' * 79)
+            lines.append('  ' + advisory.title)
+            lines.append('=' * 79)
+            for label, atr_lines in zip(labels, attributes):
+                if atr_lines in (None, [None]):
                     continue
-                for i, line in enumerate(lines):
-                    print('%*s: %s' % (width, label if i == 0 else '', line))
-            print()
+                for i, line in enumerate(atr_lines):
+                    lines.append('%*s: %s' % (width, label if i == 0 else '', line))
+            return '\n'.join(lines)
+
+        advisories = set()
+        for apkg, advisory, installed in apkg_adv_insts:
+            advisories.add(advisory2info(advisory, installed))
 
+        for advisory in sorted(advisories, key=lambda x: x.lower()):
+            print(advisory)
+            print()
```

Comment:
This will create an amty line at the end. What about: print("\n".join(sorted(advisories, key=lambda x: x.lower()))